### PR TITLE
Fix event publishing for cluster scoped crds

### PR DESCRIPTION
**Pull request by zhrebicek**
* There is no api for publishing cluster scoped events
* It default to `default` namespace
* Added ability to override namespace on Recorder::new

## Motivation

We were getting `405` when trying to publish events for cluster scoped crds.

## Solution

As there seems to be no endpoint for cluster scoped events.
I have added default namespace `default` for cluster scoped crds and I have added way to override it for all the cases.

I am not sure about the `namespace_override` but it seems good enough as is, maybe needs better comment/renaming.

cc: @teozkr

### CODE REVIEW

**Comment by zhrebicek** [comment]
I will fix the DCO  thingy and try to add it there.

@clux should I also drop the `namespace_override` or would you like to keep it there?

File: kube-runtime/src/events.rs [hunk 1]
```diff
@@ -171,12 +172,18 @@ impl Recorder {
...
-        let events = match reference.namespace.as_ref() {
-            None => Api::all(client),
-            Some(namespace) => Api::namespaced(client, namespace),
+    pub fn new(client: Client, reporter: Reporter, reference: ObjectReference, namespace_override: Option<&str>) -> Self {
+        let namespace = match namespace_override {
+            Some(namespace) => namespace,
+            None => match reference.namespace.as_ref() {
+                Some(namespace) => namespace,
+                None => "default",
+            }
```
Comment (clux) [inline kube-runtime/src/events.rs:184]:
```suggestion
    pub fn new(client: Client, reporter: Reporter, reference: ObjectReference) -> Self {
        let events = Api::namespaced(client, reference.namespace.as_ref().unwrap_or_else(|| "default");
```

if that works, might need some String::from magic

File: kube-runtime/src/events.rs [hunk 2]
```diff
@@ -171,12 +172,12 @@ impl Recorder {
     /// Create a new recorder that can publish events for one specific object
     ///
     /// This is intended to be createad at the start of your controller's reconcile fn.
```
Comment (clux) [inline kube-runtime/src/events.rs:174]:
```suggestion
    /// This is intended to be created at the start of your controller's reconcile fn.
```

File: kube-runtime/src/events.rs [hunk 3]
```diff
@@ -171,12 +172,12 @@ impl Recorder {
...
+    ///
+    /// Cluster scoped object reference defaults to "default" namespace.
```
Comment (clux) [inline kube-runtime/src/events.rs:176]:
```suggestion
    /// Cluster scoped objects will publish events in the "default" namespace.
```

**Review by clux — APPROVED** [review]
minor comment nits, but LGTM

**Pull request merged**